OpInfo: bitwise_and

Freey0 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#61349 OpInfo: bitwise_and**
* #60813 bitwise_and: Port to structured
* #60812 bitwise_xor: Port to structured
* #60811 bitwise_or: Port to strucutred
* #60817 logit_backward: Port to structured
* #60816 tanh_backward: Port to structured
* #60815 sigmoid_backward: Port to structured
* #60814 xlogy: Port to structured
* #60810 fmin has been ported to the structured kernel, removing the old implementation
* #60809 fmod: Port to structured

Also add type promotion test for bugs found by pr #60813

Differential Revision: [D29592840](https://our.internmc.facebook.com/intern/diff/D29592840)

Freey0 committed:
OpInfo: bitwise_and

Also add type promotion test for bugs found by pr #60813

[ghstack-poisoned]

ezyang commented:
@ezyang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29592840).

mruberry commented on torch/testing/_internal/common_methods_invocations.py:
```diff
@@ -1644,6 +1644,21 @@ def sample_inputs_broadcast_to(op_info, device, dtype, requires_grad, **kwargs):
             make_tensor(size, device, dtype, low=None, high=None, requires_grad=requires_grad),
             args=(shape,)) for size, shape in test_cases)
 
+def sample_inputs_bitwise_and(op_info, device, dtype, requires_grad, **kwargs):
```
Instead of a custom sample input function I think this can use

https://github.com/pytorch/pytorch/blob/10f372601dc27ccd9d1a7eaed2dbbb04fb20f962/torch/testing/_internal/common_methods_invocations.py#L1205

fyi @pmeier

mruberry commented:
Failures are unrelated - would you just switch the sample input function to the generic binary elementwise sample inputs?

Freey0 committed:
Update on "OpInfo: bitwise_and"


Also add type promotion test for bugs found by pr #60813

Differential Revision: [D29592840](https://our.internmc.facebook.com/intern/diff/D29592840)

[ghstack-poisoned]

ezyang commented:
@ezyang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29592840).

mruberry reviewed (approved):
LGTM!

Freey0 committed:
Update on "OpInfo: bitwise_and"


Also add type promotion test for bugs found by pr #60813

Differential Revision: [D29592840](https://our.internmc.facebook.com/intern/diff/D29592840)

[ghstack-poisoned]

ezyang commented:
@ezyang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29592840).
